tests/integration-tests.js

```diff
@@ -229,14 +230,11 @@ describe( 'The database', () => {
     expect( query.rowCount ).toBe( 1 );
   });
 
-  /* eslint-disable jest/expect-expect */
-  // TODO: This test really should have an assertion, but I can't figure out how to catch the error
-  //       properly... it's possible that updateScore needs rewriting to catch properly. In the
-  //       meantime, this test *does* actually work like expected.
   it( 'does not cause any errors on a second request when everything already exists', async() => {
-    await points.updateScore( defaultItem, '++' );
+    expect.hasAssertions();
+    const newScore = await points.updateScore( defaultItem, '+' );
+    expect( newScore ).toBe( 2 );
```

Comment:
Yeah, makes sense.

This adds a dependency between the first test and this one, but that's not going to cause us problems here.

I think it'd be better to verify that the _database_ has been updated, not the return value as we're checking the database integration here, not how that method works.

Reply:
In my opinion this return value is proof the database worked. Because the value is not computed by the method itself rather queried from the database after it is updated with a `set current.value +1`. If the database integration failed the return value would not have been 2 and most likely there would be an exception error.

Reply:
You're missing the point: This test is verifying that performing some action alters the database, so checking the database is the only way to verify that. We cannot trust the code to do the right thing and retrieve the result, we need to do the database call ourselves.

Reply:
Sure, I will add the validation. It is easy to do. Might as well leave nothing to chance.

Reply:
Thanks!

This is deep testing "theory" stuff: we're _technically_ in an adversarial relationship with the person writing the code, so we leave nothing to chance: Even though we all know that code is written correctly, we need to test it like it's trying to (or at least potentially might) lie to us, so we verify the expected _results_, not what the code tells us it did. So yes, adding another query here is inefficient as the code is already running it and returning the result, however we can't actually _trust_ the code to do that, so we need to run it ourselves.

Reply:
I made the changes requested.
Separate commit for clarity.